core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -556,6 +562,46 @@ public void copyLogSegmentsToRemote(UnifiedLog log) throws InterruptedException
             }
         }
 
+        public void cleanupDeletedRemoteLogSegments() {
```

Comment:
This process runs every replica. So, we will be deleting the same remote segment multiple times?

Reply:
In fact, in the 'cleanupAllRemoteLogSegments' method, it verifies whether the current node is the leader, and only the leader node will perform the deletion action.